images/virtualization-artifact/cmd/virtualization-audit/app/app.go

```
@@ -0,0 +1,107 @@
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"github.com/deckhouse/deckhouse/pkg/log"
+	"github.com/deckhouse/virtualization-controller/pkg/audit/cache"
+	"github.com/deckhouse/virtualization-controller/pkg/audit/handler"
+	"github.com/deckhouse/virtualization-controller/pkg/audit/informer"
+	"github.com/deckhouse/virtualization-controller/pkg/audit/server"
+)
+
+const long = `
+  ___  _   _______ _____ _____ ___________
+ / _ \| | | |  _  \_   _|_   _|  _  | ___ \
+/ /_\ \ | | | | | | | |   | | | | | | |_/ /
+|  _  | | | | | | | | |   | | | | | |    /
+| | | | |_| | |/ / _| |_  | | \ \_/ / |\ \
+\_| |_/\___/|___/  \___/  \_/  \___/\_| \_|
+
+`
+
+type Options struct {
+	Port     string
+	Cafile   string
+	Certfile string
+	Keyfile  string
+	Verbose  uint8
+}
+
+func NewOptions() Options {
+	return Options{}
+}
+
+func (o *Options) Flags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.Port, "secure-port", "8443", "The port to listen on")
+	fs.StringVar(&o.Cafile, "tls-ca-file", "/etc/virtualization-audit/certificate/ca.crt", "Path to TLS CA certificate")
+	fs.StringVar(&o.Certfile, "tls-cert-file", "/etc/virtualization-audit/certificate/tls.crt", "Path to TLS certificate")
+	fs.StringVar(&o.Keyfile, "tls-private-key-file", "/etc/virtualization-audit/certificate/tls.key", "Path to TLS key")
+	fs.Uint8VarP(&o.Verbose, "verbose", "v", 1, "verbose output")
+}
+
+func NewAuditCommand() *cobra.Command {
+	opts := NewOptions()
+	cmd := &cobra.Command{
```

validate no args
add silent usage and silent error